pkg/setting/setting.go

```diff
@@ -1056,6 +1059,20 @@ func readUserSettings(iniFile *ini.File, cfg *Cfg) error {
 	ViewersCanEdit = users.Key("viewers_can_edit").MustBool(false)
 	cfg.EditorsCanAdmin = users.Key("editors_can_admin").MustBool(false)
 
+	userInviteMaxLifetimeVal := valueAsString(users, "user_invite_max_lifetime_duration", "24h")
+	userInviteMaxLifetimeDuration, err := gtime.ParseInterval(userInviteMaxLifetimeVal)
+	if err != nil {
+		return err
+	}
+
+	// Minimum supported lifetime is 1 hour, and the maximum is 1 week
+	cfg.UserInviteMaxLifetime = userInviteMaxLifetimeDuration
+	if cfg.UserInviteMaxLifetime < time.Hour*1 {
+		cfg.UserInviteMaxLifetime = time.Hour * 1
+	} else if cfg.UserInviteMaxLifetime > time.Hour*24*7 {
```

Comment:
Not sure if maximum allowed is needed. What's a good reason for keeping it?

Reply:
This came up before and it was mostly about having sane values as user invites are something that should expire within `<some reasonable amount of time>`. This varies depending on your use case of Grafana / threat model of course - 1 week may be too short, but 3 months seems like a long time. I'd say what's the harm in a maximum? 😏

Reply:
We provide the default (24h) according to what we recommend is a good sane default. We shouldn't underestimate our users use cases and don't limit the maximum and let them change that to whatever they want. We've tried to limit certain things before with same reasoning as you provide and it came back and bit us eventually.

But it's no biggie really. I can live with this and instead wait and see if someone has a problem with it and then we may have to remove the limitation .

Reply:
It's a good point that we can just remove the limitation, shouldn't bother anyone if we end up doing that?

Reply:
If we are going to have a defined ranged. Then we should error if the config is outside of the range instead of changing it (since the config is not valid).